api/envoy/extensions/load_balancing_policies/subset/v3/subset.proto

```diff
@@ -0,0 +1,217 @@
+  //       { "keys": [ "stage", "hardware_type" ] }
+  //   ]}
+  //
+  // A subset is matched when the metadata from the selected route and
+  // weighted cluster contains the same keys and values as the subset's
+  // metadata. The same host may appear in multiple subsets.
+  repeated LbSubsetSelector subset_selectors = 3;
+
+  // If true, routing to subsets will take into account the localities and locality weights of the
+  // endpoints when making the routing decision.
+  //
+  // There are some potential pitfalls associated with enabling this feature, as the resulting
+  // traffic split after applying both a subset match and locality weights might be undesirable.
+  //
+  // Consider for example a situation in which you have 50/50 split across two localities X/Y
+  // which have 100 hosts each without subsetting. If the subset LB results in X having only 1
+  // host selected but Y having 100, then a lot more load is being dumped on the single host in X
+  // than originally anticipated in the load balancing assignment delivered via EDS.
+  bool locality_weight_aware = 4;
+
+  // When used with locality_weight_aware, scales the weight of each locality by the ratio
+  // of hosts in the subset vs hosts in the original subset. This aims to even out the load
+  // going to an individual locality if said locality is disproportionately affected by the
+  // subset predicate.
+  bool scale_locality_weight = 5;
+
+  // If true, when a fallback policy is configured and its corresponding subset fails to find
+  // a host this will cause any host to be selected instead.
+  //
+  // This is useful when using the default subset as the fallback policy, given the default
+  // subset might become empty. With this option enabled, if that happens the LB will attempt
+  // to select a host from the entire cluster.
+  bool panic_mode_any = 6;
+
+  // If true, metadata specified for a metadata key will be matched against the corresponding
+  // endpoint metadata if the endpoint metadata matches the value exactly OR it is a list value
+  // and any of the elements in the list matches the criteria.
+  bool list_as_any = 7;
+
+  // Fallback mechanism that allows to try different route metadata until a host is found.
+  // If load balancing process, including all its mechanisms (like
+  // :ref:`fallback_policy<envoy_v3_api_field_extensions.load_balancing_policies.subset.v3.subset.fallback_policy>`)
+  // fails to select a host, this policy decides if and how the process is repeated using another metadata.
+  //
+  // The value defaults to
+  // :ref:`METADATA_NO_FALLBACK
+  // <envoy_v3_api_field_extensions.load_balancing_policies.subset.v3.subset.LbSubsetMetadataFallbackPolicy.METADATA_NO_FALLBACK>`.
+  LbSubsetMetadataFallbackPolicy metadata_fallback_policy = 8
+      [(validate.rules).enum = {defined_only: true}];
+
+  // The child LB policy to create for endpoint-picking within the chosen subset. This child LB policy could not be
```

Comment:
Why can the child not be another subset LB policy?  IIUC, the subset LB policy will basically restrict the set of hosts to be passed to the child policy to pick from.  Shouldn't we be able to configure another subset policy to restrict that set even further before passing it down to another child?

To be clear, I don't know why someone would *want* to do that, but I also don't know of a reason why it shouldn't work.  Just trying to understand where this limitation is coming from.

Reply:
@markdroth In fact, it should works. There is no special limitation.

But the code base of subset is too complex and **this scenario is not tested**. And as you said, I think no one will want to do this. So, I just prohibit it in the API in case there are potential bug the not covered by our test.

Reply:
If there's not actually any limitation here, I think we shouldn't say that there is one.  We don't necessarily have to test all possible combinations of things people *might* do, but let's not imply that there's an artificial restriction when there actually isn't one.

Reply:
OK